JPN-549 Fixed incorrect time formatting for Japanese and Chinese dates

perjg opened the pull request:
Fixed incorrect time formatting for dates between 30 and 365 days old for Japanese and Chinese.

@Grunny This is an ad-hoc fix that works for both Japanese and Chinese, but it may not be the ideal way to fix this problem. Please see JPN-549 for my detailed comments.

perjg committed:
JPN-549 Fixed incorrect time formatting for dates between 30 and 365 days old for Japanese and Chinese

lukaszkonieczny commented on includes/wikia/GlobalFunctions.php:
```diff
@@ -937,7 +937,11 @@ function wfTimeFormatAgo( $stamp, $hideCurrentYear = true ) {
 			$format = trim( $format, ' ,yY' );
+
+			// For Japanese and Chinese the 年 character before the month number must also be stripped. See JPN-549
+			$format = str_replace('年', '', $format);
```
Maybe we could fix it globally?

perjg replied:
Not sure what you mean by globally.

lukaszkonieczny replied:
Oh, I did not see it's already in Global Functions. Right now this string replacement will be done for every language, not only Japanese or Chinese. Maybe you can investigate usage of date formats in Messages*.php files like here https://github.com/Wikia/app/blob/dev/languages/messages/MessagesZh_hans.php#L239

perjg replied:
Yes, but I think it is safe to say that str_replace with '年' will have no effect except for Japanese and Chinese.

Thanks for pointing out Messages*.php. This looks like the place to replace the default data format for Japanese and Chinese, so we can drop the (金) part from the date.

However, this fix is still required, since this is about the check for $hideCurrentYear being broken for Japanese and Chinese.

Grunny replied:
Whitespace issues. Also, couldn't you just add 年 to the trim? i.e. `trim( $format, ' ,yY年' );`

perjg committed:
JPN-549 Fix simplification

perjg committed:
JPN-549 Changed default date format for Japanese and Chinese Simplified / Traditional (Taiwan) to drop weekday name at end

perjg commented:
@Grunny @lukaszkonieczny 
I've updated the PR according to Grunny's suggestion: Using ' ,yY年' directly in the trim method works fine.

I've also changed the default date format form Japanese and Chinese Simplified / Traditional to drop the weekday name at the end.

Grunny commented:
LGTM.

perjg committed:
JPN-549 Drop weekday name at end for Nengo and Minguo date formats

perjg commented:
@Grunny Made one more commit to also drop the weekday at the end for the special Japanese Nengo and Taiwan Minguo date formats.

Grunny commented:
:+1:

perjg committed:
Merge branch 'dev' into JPN-549
